From 2637510006d406f43ae588008d9409896465ef44 Mon Sep 17 00:00:00 2001
From: ALAS Headless Research <local@alas-headless.invalid>
Date: Sat, 8 Aug 2026 20:51:00 +0800
Subject: [PATCH] null: register the Unity graphics initialization thread

---
 src/libANGLE/renderer/null/DisplayNULL.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libANGLE/renderer/null/DisplayNULL.cpp b/src/libANGLE/renderer/null/DisplayNULL.cpp
index d970c8ec1..c8db4b5fc 100644
--- a/src/libANGLE/renderer/null/DisplayNULL.cpp
+++ b/src/libANGLE/renderer/null/DisplayNULL.cpp
@@ -158,6 +158,7 @@ egl::Error DisplayNULL::initialize(egl::Display *display)
 
 #if defined(ANGLE_PLATFORM_ANDROID)
     ProbeIl2CppNamespace();
+    alas::RegisterObserverMainThread();
     alas::StartObserverServer();
 #endif
     return egl::NoError();
-- 
2.34.1
